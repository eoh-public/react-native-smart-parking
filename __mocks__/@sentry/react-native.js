export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
