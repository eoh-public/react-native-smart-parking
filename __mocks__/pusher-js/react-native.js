const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
