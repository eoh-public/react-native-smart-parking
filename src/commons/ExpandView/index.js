import { IconOutline } from '@ant-design/icons-react-native';
import React, { memo, useCallback, useState } from 'react';
import { StyleSheet, View, TouchableOpacity } from 'react-native';

import Text from '../Text';
import { Colors, Device } from '../../configs';

const ExpandView = memo(({ leftIcon, title, expandedView }) => {
  const [expanded, setExpanded] = useState(false);
  const handlePress = useCallback(() => {
    setExpanded(!expanded);
  }, [expanded]);

  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.viewClickable} onPress={handlePress}>
        {leftIcon}
        <Text type={'H4'} color={Colors.Gray9} style={styles.title}>
          {title}
        </Text>
        <IconOutline name={expanded ? 'up' : 'down'} size={15} />
      </TouchableOpacity>
      {expanded && expandedView}
    </View>
  );
});

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    marginVertical: 8,
    width: '100%',
  },
  viewClickable: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  title: {
    marginTop: !Device.isIOS ? 4 : 0,
    flex: 1,
    marginLeft: 18,
  },
  item: {
    paddingLeft: 35,
  },
  itemIcon: {
    paddingLeft: 55,
  },
  textInfo: {
    fontWeight: 'bold',
    marginTop: 10,
    marginBottom: 10,
    paddingLeft: 42,
    color: Colors.Green,
  },
  textAddCard: {
    marginTop: 10,
    marginBottom: 10,
    color: Colors.Orange,
  },
});

export default ExpandView;
