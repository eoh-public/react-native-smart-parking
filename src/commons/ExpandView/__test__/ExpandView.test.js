import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { Text, TouchableOpacity, View } from 'react-native';
import ExpandView from '../index';

describe('Test ExpandView', () => {
  const list_price = [1, 2, 3];
  let wrapper;
  let tree;

  test('create ExpandView', () => {
    act(() => {
      tree = renderer.create(
        <ExpandView
          title="Title"
          expandedView={
            <View>
              {list_price.map((price) => (
                <Text key={price} type={'H4'}>
                  {price}
                </Text>
              ))}
            </View>
          }
        />
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('expand ExpandView', () => {
    act(() => {
      wrapper = renderer.create(
        <ExpandView
          title="Title"
          expandedView={
            <View>
              {list_price.map((price) => (
                <Text key={price} type={'H4'}>
                  {price}
                </Text>
              ))}
            </View>
          }
        />
      );
    });
    const testInstance = wrapper.root;
    const button = testInstance.findByType(TouchableOpacity);
    renderer.act(() => {
      button.props.onPress();
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
