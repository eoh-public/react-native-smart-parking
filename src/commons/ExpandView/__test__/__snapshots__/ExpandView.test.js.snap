// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ExpandView create ExpandView 1`] = `
<View
  style={
    Object {
      "marginVertical": 8,
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "flex": 1,
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      Title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "antoutline",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Test ExpandView expand ExpandView 1`] = `
<View
  style={
    Object {
      "marginVertical": 8,
      "paddingHorizontal": 16,
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#262626",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 16,
            "lineHeight": 24,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {
            "flex": 1,
            "marginLeft": 18,
            "marginTop": 0,
          },
        ]
      }
    >
      Title
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontFamily": "antoutline",
            "fontSize": 15,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          undefined,
        ]
      }
    >
      
    </Text>
  </View>
  <View>
    <Text
      type="H4"
    >
      1
    </Text>
    <Text
      type="H4"
    >
      2
    </Text>
    <Text
      type="H4"
    >
      3
    </Text>
  </View>
</View>
`;
